ceed-vec.c

```diff
@@ -165,6 +174,9 @@ CEED_EXTERN int CeedVectorGetLength(CeedVector vec, CeedInt *length) {
 int CeedVectorDestroy(CeedVector *x) {
   int ierr;
 
+  if ((*x) && ((*x)->state % 2) == 1)
```

Comment:
I think this can go after the check below. It's a more permissive policy, but I don't think we need to error if one user drops their reference while another accesses the array.